src/main/java/gregtech/api/util/ShapedOreEnergyTransferRecipe.java

```diff
@@ -25,13 +25,31 @@ public ShapedOreEnergyTransferRecipe(ResourceLocation group, @Nonnull ItemStack
         this(group, result, chargePredicate, transferMaxCharge, CraftingHelper.parseShaped(recipe));
     }
 
+    public ShapedOreEnergyTransferRecipe(ResourceLocation group, @Nonnull ItemStack result, Predicate<ItemStack> chargePredicate, long chargeAmount, boolean transferMaxCharge, Object... recipe) {
+        this(group, result, chargePredicate, chargeAmount, transferMaxCharge, CraftingHelper.parseShaped(recipe));
+    }
+
     public ShapedOreEnergyTransferRecipe(ResourceLocation group, @Nonnull ItemStack result, Predicate<ItemStack> chargePredicate, boolean transferMaxCharge, ShapedPrimer primer) {
         super(group, result, primer);
         this.chargePredicate = chargePredicate;
         this.transferMaxCharge = transferMaxCharge;
         fixOutputItemMaxCharge();
     }
 
+    public ShapedOreEnergyTransferRecipe(ResourceLocation group, @Nonnull ItemStack result, Predicate<ItemStack> chargePredicate, long chargeAmount, boolean transferMaxCharge, ShapedPrimer primer) {
+        super(group, result, primer);
+        this.chargePredicate = chargePredicate;
+        this.transferMaxCharge = transferMaxCharge;
+        setMaxChargeInRecipe(chargeAmount);
+    }
```

Comment:
This is wrong you are overriding crafted item max charge to it's max charge. This will produce unnecessary NBT mess and use calculation time.
For this case we should just not call `fixOutputItemMaxCharge` which means no override which means default max energy level of item is used

Reply:
This is now done. A boolean is passed in for the new method to determine if `fixOutputItemMaxCharge` will be called, with all the old recipes having this boolean as true by default due to a redirected method.